tests/e2e/plugins/user-input.php

```diff
@@ -0,0 +1,139 @@
+<?php
+/**
+ * Plugin Name: E2E Tests User Input Settings API Mock
+ * Plugin URI:  https://github.com/google/site-kit-wp
+ * Description: Utility plugin for handling user input settings for Site Kit during E2E tests.
+ * Author:      Google
+ * Author URI:  https://opensource.google.com
+ *
+ * @package   Google\Site_Kit
+ * @copyright 2019 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://sitekit.withgoogle.com
+ */
+
+use Google\Site_Kit\Context;
+use Google\Site_Kit\Core\Authentication\Google_Proxy;
+use Google\Site_Kit\Core\Authentication\User_Input_State;
+use Google\Site_Kit\Core\Storage\User_Options;
+
+register_activation_hook( __FILE__, 'e2e_user_input_settings_reset' );
+register_deactivation_hook( __FILE__, 'e2e_user_input_settings_reset' );
+
+add_filter( 'pre_http_request', 'e2e_user_input_settings_proxy_handler', 10, 3 );
+
+function e2e_user_input_settings_reset() {
+	$user_options     = new User_Options( new Context( GOOGLESITEKIT_PLUGIN_MAIN_FILE ) );
+	$user_input_state = new User_Input_State( $user_options );
```

Comment:
While this works fine for the tests, we should account for this potentially being activated (or deactivated) with Site Kit inactive. Other E2E plugins check for a constant or use Site Kit provided actions to prevent such errors.

Maybe we should `wp_die` on `plugins_loaded` if Site Kit isn't active? I'm open to ideas here.

Reply:
Updated to check whether the `GOOGLESITEKIT_PLUGIN_MAIN_FILE` constant is defined.